docs/pluginconfig.md

```diff
@@ -0,0 +1,39 @@
+# Initiating Plugin - Versioning model
+
+When initiating plugin code and calling `GenerateConfig` method we need to
+pass in `Plugin Template/PluginConfig` as a parameter. 
+
+Plugin template defines main characteristics for the cluster. In example 
+cluster, images versions, VM image, external integration details.
+
+Config file examples can be found in `pluginconfig/pluginconfig-{version}.yaml`, but ideally config object should be constructed using `pkg/api/plugin/config.go Config` struct.
+
+`openshift-azure` repository code always contains stable newest version of file references.
+
+Before updating example file and code base for new release aka. code cut for MSFT RP, make sure you tag code with corresponding release tag version
```

Comment:
I'm struggling to understand what the topic of the doc is from this point forwards.  What do you mean by tags here?